charts/argo-cd/templates/argocd-server/certificate.yaml

```diff
@@ -16,14 +16,14 @@ metadata:
   labels:
     {{- include "argo-cd.labels" (dict "context" . "component" .Values.server.name "name" .Values.server.name) | nindent 4 }}
 spec:
-  commonName: {{ .Values.server.certificate.domain | quote }}
+  commonName: {{ (include "coreweave.externalDnsName" .) | quote }}
   dnsNames:
-    - {{ .Values.server.certificate.domain | quote }}
+    - {{ (include "coreweave.externalDnsName" . ) | quote }}
   {{- range .Values.server.certificate.additionalHosts }}
     - {{ . | quote }}
   {{- end }}
   issuerRef:
-    kind: {{ .Values.server.certificate.issuer.kind | quote }}
-    name: {{ .Values.server.certificate.issuer.name | quote }}
-  secretName: {{ .Values.server.certificate.secretName | quote }}
+    kind: ClusterIssuer
+    name: letsencrypt-prod
+  secretName: "argocd-tls-cert"
```

Comment:
Secret name should include Release.Name or some randomness so multiple could be deployed. Not that it's likely someone will need 2 argocds, but better to plan for that niche case.